ClientApp/KachnaOnline/src/app/states/plan-state/plan-state.component.ts

```diff
@@ -25,83 +35,135 @@ export class PlanStateComponent implements OnInit {
     notePublic: new FormControl('')
   });
 
-  planningNew: boolean = false;
+  editingId: number;
+  mode: Mode;
+  referenceState: ClubState;
 
   constructor(public stateService: StatesService,
               public calendar: NgbCalendar,
               private nativeDateAdapter: NgbDateNativeAdapter,
               private route: ActivatedRoute,
-              private router: Router) { }
+              private router: Router) {
+  }
 
   ngOnInit(): void {
-    this.planningNew = this.route.snapshot.data.planningNew ?? true;
-    this.stateService.getCurrent().subscribe(currentState => {
-      if (!this.planningNew && currentState.state != ClubStateTypes.Closed) {
-        this.mainForm.controls.plannedEndDate.setValue(this.nativeDateAdapter.fromModel(currentState.plannedEnd));
+    const stateFetched = (state: ClubState) => {
+      this.referenceState = state;
+
+      if (this.mode == Mode.ModifyCurrent && state.state == ClubStateTypes.Closed) {
+        this.mode = Mode.CreateCurrent;
+      }
+
+      if (this.mode != Mode.CreateCurrent && this.mode != Mode.CreatePlanned && state.state != ClubStateTypes.Closed) {
+        // TODO: Is there a better, more Angular-ish way to do this mapping?
```

Comment:
Yes, there is. You can use `patchValue` on `FromGroup` and pass in an object with a structure that follows the form and it will be updated so you don't have to set each control manually